core/test/com/treode/async/misc/DirectDequeSpec.scala

```diff
@@ -0,0 +1,139 @@
+package com.treode.async.misc
+
+import org.scalatest.FlatSpec
+
+class DirectDequeSpec extends FlatSpec {
+
+  def assertElements (es: Int*) (deque: DirectDeque[Int]) {
+    if (es.isEmpty) {
+      assert (deque.isEmpty)
+
+      intercept[NoSuchElementException] { deque.get(0) }
+      intercept[NoSuchElementException] { deque.dequeue() }
+    } else {
+      assert (deque.size == es.size)
+      assert (Seq.tabulate (deque.size) (deque.get (_)) == es)
+
+      intercept[NoSuchElementException] { deque.get(-1) }
+      intercept[NoSuchElementException] { deque.get(deque.size) }
+    }
+  }
+
+  "DirectDeque" should "nq more than the deque initial capacity" in {
+    val capacity = 3
+    val deque = new DirectDeque[Int](capacity)
+    // enqueue more than the initial capacity
+    for (i <- 1 to capacity+1) {
+      deque.enqueue(i)
+    }
+    assertElements (1, 2, 3, 4) (deque)
+  }
+
+  it should "err, nq, dq, err" in {
+    val deque = new DirectDeque[Int](1)
+    assertElements () (deque)
+    // enqueue one element
```

Comment:
This comment says what the next line of code says. It's not helpful and can be removed.